portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,20 +56,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/walkthrough/");
   }
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
+  private int getCommentLimit(String commentLimitString) {
+    int commentLimit;
+    try {
+      commentLimit = Integer.parseInt(commentLimitString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + commentLimitString);
+      return -1;
+    }
+    if (commentLimit < 0) {
+      System.err.println("Comment limit is out of range: " + commentLimitString);
+      return -1;
+    }
+    return commentLimit;
+  }
+
+  private List<String> getComments(int commentLimit) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     List<String> comments = new ArrayList<String>();
 
+    int commentsAdded = 0;
     for (Entity entity : results.asIterable()) {
+      if (commentsAdded >= commentLimit) {
+        break;
+      }
       String content = (String) entity.getProperty("content");
       comments.add(content);
+      commentsAdded++;
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentLimitString = request.getParameter("comment_limit");
+    int commentLimit = getCommentLimit(commentLimitString);
```

Comment:
i also think this doGet method would be cleaner if you passed the parameter directly to the getComments method, and called getCommentLimit from within there.